### PR TITLE
[Fix #3480] Changing jsonb to varchar

### LINKED ISSUE #3480: add correlation persistence support for mssql
### Description

save the correlation ids from cloudevent in db and use them to lookup existing process instance to resume execution

### Implementation ideas

add an implementation for mssql similar to PostgreSQLCorrelationService

### PR DESCRIPTION
**Pull request by fjtirado**
Fixes https://github.com/apache/incubator-kie-kogito-runtimes/issues/3480

### CODE REVIEW

**Comment by kie-ci3** [comment]
**PR job** `#7` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3484 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3484/7/display/redirect

**Test results:**
- PASSED: 2867
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3484/7/testReport/org.kie.kogito.mongodb.transaction/AbstractTransactionManagerIT/testAbort/">org.kie.kogito.mongodb.transaction.AbstractTransactionManagerIT.testAbort</a></summary>
Command failed with error 246 (SnapshotUnavailable): 'Unable to read from a snapshot due to pending collection catalog changes; please retry the operation. Snapshot timestamp is Timestamp(1713883869, 0). Collection minimum is Timestamp(1713883869, 2)' on server 172.18.0.1:32799. The full response is {"errorLabels": ["TransientTransactionError"], "operationTime": {"$timestamp": {"t": 1713883868, "i": 6}}, "ok": 0.0, "errmsg": "Unable to read from a snapshot due to pending collection catalog changes; please retry the operation. Snapshot timestamp is Timestamp(1713883869, 0). Collection minimum is Timestamp(1713883869, 2)", "code": 246, "codeName": "SnapshotUnavailable", "$clusterTime": {"clusterTime": {"$timestamp": {"t": 1713883869, "i": 3}}, "signature": {"hash": {"$binary": {"base64": "AAAAAAAAAAAAAAAAAAAAAAAAAAA=", "subType": "00"}}, "keyId": 0}}}
</details>

**Comment by deepakkapoor23** [comment]
This doesn't really address #3469, only affects JDBC persistence...but that's enough for my use case right now. Thanks @fjtirado

**Review by nmirasch — APPROVED** [review]

**Comment by fjtirado** [comment]
> This doesn't really address #3469, only affects JDBC persistence...but that's enough for my use case right now. Thanks @fjtirado

You are right, MongoDB is not relational ;)

File: addons/common/persistence/jdbc/src/main/resources/db/ansi/V1.35.0__create_runtime_ansi.sql [hunk 1]
```diff
@@ -1,10 +1,10 @@
 CREATE TABLE process_instances
 (
-    id              CHAR(36)      NOT NULL,
-    payload         BLOB          NOT NULL,
-    process_id      VARCHAR(4000) NOT NULL,
-    version         BIGINT(19),
-    process_version VARCHAR(4000),
+    id              character(36)      NOT NULL,
+    payload         varbinary(1000000) NOT NULL,
```
Comment (gmunozfe) [inline addons/common/persistence/jdbc/src/main/resources/db/ansi/V1.35.0__create_runtime_ansi.sql:4]: I guess you rounded up from a blob that it's strictly 65535 bytes.

**Review by gmunozfe — APPROVED** [review]
Looks good to me, well done @fjtirado !

File: addons/common/persistence/jdbc/src/main/resources/db/ansi/V1.35.0__create_runtime_ansi.sql (hunk 1 above)
Comment (fjtirado) [reply addons/common/persistence/jdbc/src/main/resources/db/ansi/V1.35.0__create_runtime_ansi.sql:4]: yes, I put a ridicolous high value, hoping that it exceeds the max allowed by an implementation, it wont fail and will be capped to the max. In any case, it is more "standard" than blob (which is supported only by oracle and mysql)

**Pull request merged**